crates/teloxide/examples/deep_linking.rs

```diff
@@ -0,0 +1,133 @@
+use dptree::{case, deps};
+use teloxide::{
+    dispatching::dialogue::{self, InMemStorage},
+    macros::BotCommands,
+    prelude::*,
+    types::Me,
+};
+
+pub type MyDialogue = Dialogue<State, InMemStorage<State>>;
+pub type HandlerResult = Result<(), Box<dyn std::error::Error + Send + Sync>>;
+
+#[derive(Clone, PartialEq, Debug, Default)]
+pub enum State {
+    #[default]
+    Start,
+    WriteToSomeone {
+        id: i64,
+    },
+}
+
+#[derive(BotCommands, Clone, Debug)]
+#[command(rename_rule = "lowercase")]
+pub enum StartCommand {
+    #[command()]
+    Start(String), /* Because deep linking (links like https://t.me/some_bot?start=123456789)
+                    * is the same as sending "/start 123456789",
+                    * we can treat it as just an argument to a command
+                    *
+                    * https://core.telegram.org/bots/features#deep-linking */
+}
+
+#[tokio::main]
+async fn main() {
+    pretty_env_logger::init();
+    log::info!("Starting dialogue bot...");
+
+    let bot = Bot::from_env();
+
+    let handler = dialogue::enter::<Update, InMemStorage<State>, State, _>()
+        .branch(
+            Update::filter_message()
+                .filter_command::<StartCommand>() // Nessary to get cmd as an argument
+                .branch(case![StartCommand::Start(start)].endpoint(start)),
+        )
+        .branch(
+            Update::filter_message()
+                .branch(case![State::WriteToSomeone { id }].endpoint(send_message)),
+        );
+
+    Dispatcher::builder(bot, handler)
+        .dependencies(deps![InMemStorage::<State>::new()])
+        .enable_ctrlc_handler()
+        .build()
+        .dispatch()
+        .await;
+}
+
+pub async fn start(
+    bot: Bot,
+    dialogue: MyDialogue,
+    msg: Message,
+    cmd: StartCommand,
+    me: Me,
+) -> HandlerResult {
+    // If you have multiple commands, this will need to become a match
```

Comment:
or just several branches in the handler tree

Reply:
What?
```
let StartCommand::Start(arg) = cmd;
```
Will need to become smth like 
```
let StartCommand::Start(arg) = cmd else {
    unreachable!()
};
```
If there becomes a second command like /cancel, handler tree is not in play here

Reply:
If you filter `/start` command in your handler tree, you know for sure what it's argument is. Just another way, not necessary to do it here, as long as there is one command

Reply:
Yes, yes, but rust will complain if there is a second command, that's why i added that comment